src/poller.js

```diff
@@ -35,6 +35,12 @@ module.exports = EventEmitter => {
 			if (config.autostart) {
 				this.start ({initialRequest: true});
 			}
+
+			// We must listen to the error event to prevent throwing when we receive an HTTP error
+			// https://nodejs.org/docs/latest-v16.x/api/events.html#error-events
+			this.on('error', (error) => {
+				this.error = error;
```

Comment:
**question**: why not set `this.error` in the `catch` on [line 108](https://github.com/Financial-Times/ft-poller/pull/91/files#diff-f961eaba5259cc41eb79468c42aa5697d57c0ca436c7da8f92da4dcc958f1600R108) rather than emitting an `error` event then listening for it in the same object?

Reply:
I _think_ (Kara can correct me if she had other ideas) we didn't want to remove the `emit('error')` because some clients may be listening for it, but we do need to listen for it at least once otherwise the emit inside the catch will re-throw.

Reply:
That makes sense thank you, I forgot you could have multiple listeners for an event rather than the first one consuming it.